lib/iris/coords.py

```diff
@@ -2775,6 +2779,366 @@ def xml_element(self, doc):
+
+    @property
+    def has_equal_element_lengths(self):
+        """
+        A boolean stating whether all values from
+        :attr:`element_lengths` are equal. Provided
+        for lazy calculation without needing to realise :attr:`indices` or
+        :attr:`element_lengths`.
+
+        """
+        lengths = self._lazy_element_lengths(self.indices, self.element_dim)
+        has_equal = lengths.min() == lengths.max()
+        return has_equal.compute()
+
+    def _validate_indices(self, indices):
+        def indices_error(message):
+            raise ValueError("Invalid indeces provided. " + message)
+
+        indices = self._sanitise_array(indices, 0)
+
+        indices_dtype = indices.dtype
+        if not np.issubdtype(indices_dtype, np.integer):
+            indices_error(
+                f"dtype must be numpy integer subtype, got: {indices_dtype} ."
+            )
+
+        indices_min = indices.min()
+        if _lazy.is_lazy_data(indices_min):
+            indices_min = indices_min.compute()
+        if indices_min < self.start_index:
+            indices_error(
+                f"Lowest index: {indices_min} < start_index: {self.start_index} ."
+            )
+
+        indices_shape = indices.shape
+        if len(indices_shape) != 2:
+            indices_error(
+                f"Expected 2-dimensional shape, got: shape={indices_shape} ."
+            )
+
+        len_req_fail = False
+        element_lengths = self._lazy_element_lengths(indices, self.element_dim)
+        if self.cf_role_element in ("edge", "boundary"):
+            if (element_lengths != 2).any().compute():
+                len_req_fail = "len=2"
+        else:
+            if self.cf_role_element == "face":
+                min_size = 3
+            elif self.cf_role_element == "volume":
+                if self.cf_role_indexed_element == "edge":
+                    min_size = 6
+                else:
+                    min_size = 4
+            else:
+                raise NotImplementedError
+            if (element_lengths < min_size).any().compute():
+                len_req_fail = f"len>={min_size}"
+        if len_req_fail:
+            indices_error(
+                f"Not all elements meet requirement: {len_req_fail} - needed "
+                f"to describe '{self.cf_role}' ."
+            )
+
+    def switch_start_index(self):
+        # Don't use a normal property setter - would be inappropriate since
+        # other properties are also modified.
+        """
+        Change :attr:`start_index` to its alternative value and change the
+        :attr:`indices` values to match. (:attr:`start_index` can be either
+        ``0`` or ``1``).
+
+        """
+        new_index = 1 - self.start_index
+        index_change = new_index - self.start_index
+        self._metadata_manager.start_index = new_index
+        # Modify the indices values to correspond with the new start_index.
+        self._values += index_change
```

Comment:
This is a complete deal breaker for me.

The `Connectivity` doesn't live in isolation, it's used to index into the associated mesh coordinates. Changing the `start_index` and then using it will result in something very broken.

Reply:
97310e758a5b23a92c28910d885bfaa2aa2c79dd